src/examples/LottieSlot.cpp

```diff
@@ -0,0 +1,227 @@
+/*
+ * Copyright (c) 2024 the ThorVG project. All rights reserved.
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include "Common.h"
+
+/************************************************************************/
+/* Drawing Commands                                                     */
+/************************************************************************/
+
+static unique_ptr<tvg::LottieAnimation> animation;
+static Elm_Transit *transit;
+static bool updated = false;
+
+void tvgUpdateCmds(tvg::Canvas* canvas, tvg::LottieAnimation* animation, float progress)
+{
+    if (updated || !canvas) return;
+
+    //Update animation frame only when it's changed
+    if (animation->frame(animation->totalFrame() * progress) == tvg::Result::Success) {
+        canvas->update();
+        updated = true;
+    }
+}
+
+void tvgDrawCmds(tvg::Canvas* canvas)
+{
+    //Animation Controller
+    animation = tvg::LottieAnimation::gen();
+    auto picture = animation->picture();
+
+    //Background
+    auto shape = tvg::Shape::gen();
+    shape->appendRect(0, 0, WIDTH, HEIGHT);
+    shape->fill(50, 50, 50);
+
+    if (canvas->push(std::move(shape)) != tvg::Result::Success) return;
+
+    const char* slotData = R"(
+    {
+        "gradient_fill": {
+            "p": {
```

Comment:
this json is ok?

Reply:
<img width="910" alt="Screenshot 2024-02-21 at 1 00 42 AM" src="https://github.com/thorvg/thorvg/assets/11167117/f5341206-6a5b-441a-abc9-5ca3a894a5f7">
Yes, In some white-space cases, RapidJSON cannot parse, but this example JSON can be parsed.

We might need to find the crashed pattern with the white-space.